patches/minecraft/net/minecraft/tags/Tag.java.patch

```diff
@@ -0,0 +1,60 @@
+--- a/net/minecraft/tags/Tag.java
++++ b/net/minecraft/tags/Tag.java
+@@ -118,8 +118,11 @@
+       }
+ 
+       public boolean func_200160_a(Function<ResourceLocation, Tag<T>> p_200160_1_) {
+-         for(Tag.ITagEntry<T> itagentry : this.field_200052_a) {
++         java.util.Iterator<Tag.ITagEntry<T>> iterator = this.field_200052_a.iterator();
++         while (iterator.hasNext()) {
++            Tag.ITagEntry<T> itagentry = iterator.next();
+             if (!itagentry.func_200161_a(p_200160_1_)) {
++               if (itagentry.isOptional()) { iterator.remove(); continue; }
+                return false;
+             }
```

Comment:
this is Tag.Builder's resolve method. Normally if any entry fails to resolve false is returned, but we insert a check to instead remove the entry and keep going if the entry is "optional"